chapter8.adoc

```diff
@@ -30,8 +70,45 @@ security manager using secure memory.
 When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
 `MSDCFG.sdedbgalw` is 1, then the external debug mode may be entered but is
 restricted by `M-mode` software to prevent privileged CSR accesses and memory
-accesses by instructions executed in external debug mode cannot use `M` privilege.
-When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, the M-mode root domain
-security manager must not configure triggers with `action=1`. Triggers for the
-supervisor domain are expected to be controlled by the root domain security
-manager to prevent any leakage of information.
+accesses by instructions executed in Debug Mode cannot use `M-mode`
+privilege. When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, the `M-mode`
+root domain security manager must not configure triggers with `action=1`.
+Triggers for the supervisor domain must be controlled by the root
+domain security manager to prevent any leakage of information across security domains.
+
+=== `Smsdetrc`: External Trace allowed control for Supervisor Domain
+
+When M-mode external trace is enabled, all supervisor domains activity may also
+be traced by an external debugger irrespective of the configuration held in
+`msdcfg.SDETRCALW`.
+
+When M-mode external trace is disabled, whether execution at privilege modes
+less than `M-mode` may be traced by an external debugger depends on the
+configuration held in `msdcfg.SDETRCALW`.
+
+When `msdcfg.SDETRCALW` = 0, external trace is disallowed for the SD. All trace
+output from the trace encoder module are disabled while the SD is active (the
+trace encoder must behave per `trTeEnable`=0).
+
+When `msdcfg.SDEDBGALW` = 1 then external trace of privilege modes less than
+`M-mode` is allowed for the SD, and programming of `trTeActive` and `trTeEnable`
+to 1 may transition the hart to enable trace for all privilege modes less than
+`M-mode`.
+
+==== `Smsdetrc`: interaction with external trace security controls (Informative)
+
```

Comment:
Why not:
When metrcen is 0, halted is asserted on:
  - Entry to Debug mode.
  - Entry to privilege M.
  - Entry to privilege < M with sdetrcalw == 0.
When metrcen is 0, halted is deasserted on:
  - Entry to privilege < M with sdetrcalw == 0.
When metrcen is 1, halted is asserted on:
  - Entry to Debug mode.
When metrcen is 1, halted is deasserted on:
  - Resume from Debug mode.

Reply:
thats the same as the following 

> When `metrcen` is 0 and privilege is `M-mode`, `M-mode` activity cannot be
> traced and `M-mode` software can enforce that `trTeEnable` is set to the state
> of `msdcfg.SDETRCALW` for the SD it context switches the hart into.

Reply:
ack